Date: 2014/09/23 12:32:57, Class: Database, Function: connect, Error: com.mysql.jdbc.CommunicationsException: Communications link failure due to underlying exception: 

** BEGIN NESTED EXCEPTION ** 

java.net.ConnectException
MESSAGE: Connection refused: connect

STACKTRACE:

java.net.ConnectException: Connection refused: connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:339)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:200)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:182)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:579)
	at java.net.Socket.connect(Socket.java:528)
	at java.net.Socket.<init>(Socket.java:425)
	at java.net.Socket.<init>(Socket.java:241)
	at com.mysql.jdbc.StandardSocketFactory.connect(StandardSocketFactory.java:256)
	at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:271)
	at com.mysql.jdbc.Connection.createNewIO(Connection.java:2771)
	at com.mysql.jdbc.Connection.<init>(Connection.java:1555)
	at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:285)
	at java.sql.DriverManager.getConnection(DriverManager.java:571)
	at java.sql.DriverManager.getConnection(DriverManager.java:215)
	at CorpseSlasherServer.Database.connect(Database.java:41)
	at CorpseSlasherServer.DatabaseUpdate.checkUsernameAvailable(DatabaseUpdate.java:97)
	at CorpseSlasherServer.Input.getInput(Input.java:143)
	at CorpseSlasherServer.ClientConnection.run(ClientConnection.java:47)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)


** END NESTED EXCEPTION **



Last packet sent to the server was 1 ms ago.
Date: 2014/09/23 12:32:57, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/23 12:33:00, Class: Database, Function: connect, Error: com.mysql.jdbc.CommunicationsException: Communications link failure due to underlying exception: 

** BEGIN NESTED EXCEPTION ** 

java.net.ConnectException
MESSAGE: Connection refused: connect

STACKTRACE:

java.net.ConnectException: Connection refused: connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:339)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:200)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:182)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:579)
	at java.net.Socket.connect(Socket.java:528)
	at java.net.Socket.<init>(Socket.java:425)
	at java.net.Socket.<init>(Socket.java:241)
	at com.mysql.jdbc.StandardSocketFactory.connect(StandardSocketFactory.java:256)
	at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:271)
	at com.mysql.jdbc.Connection.createNewIO(Connection.java:2771)
	at com.mysql.jdbc.Connection.<init>(Connection.java:1555)
	at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:285)
	at java.sql.DriverManager.getConnection(DriverManager.java:571)
	at java.sql.DriverManager.getConnection(DriverManager.java:215)
	at CorpseSlasherServer.Database.connect(Database.java:41)
	at CorpseSlasherServer.DatabaseUpdate.setNewUser(DatabaseUpdate.java:38)
	at CorpseSlasherServer.Input.getInput(Input.java:57)
	at CorpseSlasherServer.ClientConnection.run(ClientConnection.java:47)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)


** END NESTED EXCEPTION **



Last packet sent to the server was 0 ms ago.
Date: 2014/09/23 12:33:00, Class: Database, Function: addUser, Error: java.lang.NullPointerException
Date: 2014/09/23 12:33:06, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:10:38, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:10:38, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:10:59, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:11:53, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:12:50, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:12:50, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:13:19, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:13:19, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 10:13:19, Class: DatabaseUpdate, Function: retriveLeaderBoard, Error: java.lang.NullPointerException
Date: 2014/09/26 10:13:21, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:13:21, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 10:13:21, Class: DatabaseUpdate, Function: retriveLeaderBoard, Error: java.lang.NullPointerException
Date: 2014/09/26 10:13:24, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:15:06, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:15:06, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:17:52, Class: Input, Function: getInput, Error: org.json.JSONException: JSONObject["username"] not found.
Date: 2014/09/26 10:19:12, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:22:59, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:22:59, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:23:20, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:32:28, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:32:28, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:33:09, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:38:29, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:38:29, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:40:27, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:41:43, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:41:43, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:42:17, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:43:40, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:45:15, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:45:15, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:45:32, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:47:36, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:47:36, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:48:16, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:53:58, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:53:58, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:54:35, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:55:20, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:55:20, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:55:53, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 10:56:21, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 10:56:21, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 10:57:16, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:12:47, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:12:47, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 11:13:47, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:14:17, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:14:17, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 11:14:36, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:14:36, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 11:14:36, Class: DatabaseUpdate, Function: retriveLeaderBoard, Error: java.lang.NullPointerException
Date: 2014/09/26 11:14:59, Class: Input, Function: getInput, Error: org.json.JSONException: JSONObject["username"] not found.
Date: 2014/09/26 11:16:22, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:24:41, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:24:41, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 11:24:58, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:24:58, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 11:24:58, Class: DatabaseUpdate, Function: retriveLeaderBoard, Error: java.lang.NullPointerException
Date: 2014/09/26 11:25:22, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:25:51, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:25:51, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 11:26:22, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:26:22, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 11:26:22, Class: DatabaseUpdate, Function: retriveLeaderBoard, Error: java.lang.NullPointerException
Date: 2014/09/26 11:26:47, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:27:50, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:27:50, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 11:28:39, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:28:39, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 11:28:39, Class: DatabaseUpdate, Function: retriveLeaderBoard, Error: java.lang.NullPointerException
Date: 2014/09/26 11:28:58, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:29:46, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:29:46, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 11:30:08, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:30:08, Class: Database, Function: addUser, Error: java.lang.NullPointerException
Date: 2014/09/26 11:30:57, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:33:56, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:33:56, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 11:34:08, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:34:08, Class: Database, Function: addUser, Error: java.lang.NullPointerException
Date: 2014/09/26 11:34:22, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:34:59, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:34:59, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 11:35:00, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:35:00, Class: Database, Function: addUser, Error: java.lang.NullPointerException
Date: 2014/09/26 11:35:13, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:35:31, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:35:31, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 11:35:32, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:35:32, Class: Database, Function: addUser, Error: java.lang.NullPointerException
Date: 2014/09/26 11:36:19, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:36:19, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 11:36:56, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:37:08, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:37:08, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 11:40:37, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:40:37, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 11:41:57, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:41:57, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 11:41:57, Class: DatabaseUpdate, Function: retriveLeaderBoard, Error: java.lang.NullPointerException
Date: 2014/09/26 11:42:35, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:42:35, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 11:42:57, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:42:57, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 11:42:57, Class: DatabaseUpdate, Function: retriveLeaderBoard, Error: java.lang.NullPointerException
Date: 2014/09/26 11:43:28, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:43:28, Class: Database, Function: login, Error: java.lang.NullPointerException
Date: 2014/09/26 11:44:08, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:45:33, Class: Database, Function: connect, Error: java.sql.SQLException: Access denied for user 'root'@'localhost' (using password: YES)
Date: 2014/09/26 11:45:33, Class: Database, Function: availableUsername, Error: java.lang.NullPointerException
Date: 2014/09/26 11:45:33, Class: DatabaseUpdate, Function: retriveLeaderBoard, Error: java.lang.NullPointerException
Date: 2014/09/26 11:46:46, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:50:15, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:52:10, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/09/26 11:53:29, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/10/06 15:27:35, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/10/06 15:36:17, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
Date: 2014/10/06 15:37:02, Class: ClientConnection, Function: run, Error: java.net.SocketException: Connection reset
